synapse/lib/stormlib/model.py

```diff
@@ -1269,3 +1020,266 @@ async def _riskHasVulnToVulnerable(self, n, nodata=False):
+                    (s_layer.EDIT_PROP_SET, (propname, propvalu, None, stortype), ()),
+                )
+
+        for layriden, sode in node['sodes'].items():
+            nodeedits.setdefault(layriden, {})
+            nodeedits[layriden].setdefault(buid, (buid, nodeform, []))
+
+            for propname, propvalu in sode.get('props', {}).items():
+                propvalu, stortype = propvalu
+
+                nodeedits[layriden][buid][2].append(
+                    (s_layer.EDIT_PROP_SET, (propname, propvalu, None, stortype), ()),
+                )
+
+            for tagname, tagvalu in sode.get('tags', {}).items():
+                nodeedits[layriden][buid][2].append(
+                    (s_layer.EDIT_TAG_SET, (tagname, tagvalu, None), ()),
+                )
+
+            for tagprop, tagpropvalu in sode.get('tagprops', {}).items():
+                for propname, propvalu in tagpropvalu.items():
+                    propvalu, stortype = propvalu
+                    nodeedits[layriden][buid][2].append(
+                        (s_layer.EDIT_TAGPROP_SET, (tagname, propname, propvalu, None, stortype), ()),
+                    )
+
+        for layriden, data in node['nodedata'].items():
+            nodeedits.setdefault(layriden, {})
+            nodeedits[layriden].setdefault(buid, (buid, nodeform, []))
+
+            for name, valu in data:
+                nodeedits[layriden][buid][2].append(
+                    (s_layer.EDIT_NODEDATA_SET, (name, valu, None), ()),
+                )
+
+        for layriden, edges in node['n1edges'].items():
+            nodeedits.setdefault(layriden, {})
+            nodeedits[layriden].setdefault(buid, (buid, nodeform, []))
+
+            for verb, iden in edges:
+                nodeedits[layriden][buid][2].append(
+                    (s_layer.EDIT_EDGE_ADD, (verb, iden), ()),
+                )
+
+        for layriden, edges in node['n2edges'].items():
+            n1iden = s_common.ehex(buid)
+
+            for verb, iden, n2form in edges:
+                n2buid = s_common.uhex(iden)
+
+                nodeedits.setdefault(layriden, {})
+                nodeedits[layriden].setdefault(n2buid, (n2buid, n2form, []))
+
+                nodeedits[layriden][n2buid][2].append(
+                    (s_layer.EDIT_EDGE_ADD, (verb, n1iden), ()),
+                )
+
+        for layriden, reflist in node['refs'].items():
+            for iden, refinfo in reflist:
+                formname, propname, proptype, isarray, isro = refinfo
+
+                if isro:
+                    continue
+
+                refbuid = s_common.uhex(iden)
+
+                nodeedits.setdefault(layriden, {})
+                nodeedits[layriden].setdefault(refbuid, (refbuid, formname, []))
+
+                stortype = self.runt.snap.core.model.type(proptype).stortype
+
+                nodeedits[layriden][refbuid][2].append(
+                    (s_layer.EDIT_PROP_SET, (propname, norm, None, stortype), ()),
```

Comment:
Does this need to check the ref type to handle ndef/array refs differently?

Reply:
yes, good catch! I'll update.